language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/BallerinaSymbolServiceImpl.java

```diff
@@ -41,4 +68,74 @@
         return endpoints;
     }
 
+    @Override
+    public CompletableFuture<ExpressionTypeResponse> type(ExpressionTypeRequest request) {
+        return CompletableFuture.supplyAsync(() -> {
+            ExpressionTypeResponse expressionTypeResponse = new ExpressionTypeResponse();
+            String fileUri = request.getDocumentIdentifier().getUri();
+            Optional<Path> filePath = CommonUtil.getPathFromURI(fileUri);
+            if (!filePath.isPresent()) {
+                return expressionTypeResponse;
+            }
+
+            try {
+                expressionTypeResponse.setDocumentIdentifier(new TextDocumentIdentifier(fileUri));
+
+                // Get the semantic model.
+                Optional<SemanticModel> semanticModel = this.workspaceManager.semanticModel(filePath.get());
+
+                if (semanticModel.isPresent()) {
+                    Optional<Symbol> symbol = Optional.empty();
+                    for (int i = 0; i < request.getPosition().offset(); i++) {
+                        symbol = semanticModel.get().symbol(workspaceManager.document(filePath.get()).get(), LinePosition.from(request.getPosition().line(), request.getPosition().offset() - i));
+                        if (symbol.isPresent()) {
+                            break;
+                        }
+                    }
+                    if (symbol.isPresent()) {
+                        expressionTypeResponse.setTypes(getAllUnionTypes(symbol.get()));
```

Comment:
Do we get only the union types here? If so the API's name is misleading. Better we have the generic name for the API (as now) and flag the type/ add the type kind which can be used to filter the unions during the usage

Reply:
No, we get all types here. It should be `getAllTypes()`. I'll rename it.